clients/src/main/java/org/apache/kafka/clients/NetworkClientUtils.java

```diff
@@ -65,7 +65,7 @@ public static boolean awaitReady(KafkaClient client, Node node, Time time, long
             return true;
 
         long attemptStartTime = time.milliseconds();
-        while (!client.isReady(node, attemptStartTime) && attemptStartTime < expiryTime) {
+        while (!client.isReady(node, attemptStartTime) && attemptStartTime - expiryTime < 0) {
```

Comment:
Would it make sense to create a utility function for this given it's used in 9 places?